File: stackflow-frontend/src/app/models/project.ts
export type ProjectStatus = 'Planned' | 'Active' | 'Completed' | 'OnHold';
export interface ProjectEmployee { id: number; name: string; }
export interface Project {
  id: number; name: string; description: string; startDate: string;
  endDate: string | null; status: string; employees: ProjectEmployee[];
}
export interface ProjectRequest {
  name: string; description: string; startDate: string;
  endDate: string | null; status: ProjectStatus;
}

File: stackflow-api/Models/Project.cs
namespace stackflow_api.Models;

public class Project
{
    public int Id { get; set; }
    public string Name { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;
    public DateOnly StartDate { get; set; }
    public DateOnly? EndDate { get; set; }
    public ProjectStatus Status { get; set; }
    public ICollection<EmployeeProject> EmployeeProjects { get; set; } = [];
}
